jdaviz/configs/default/plugins/model_fitting/model_fitting.py

```diff
@@ -801,3 +809,34 @@ def _register_spectrum(self, event):
 
         self.add_results.add_results_from_plugin(spectrum)
         self._set_default_results_label()
+
+    def _apply_subset_masks(self, spectrum, dimensions=['spectral', 'spatial']):
+        """
+        Iterate over dimensions of a spectral cube ``spectrum``, add a mask attribute
+        if none exists. Mask excludes non-selected spectral and/or spatial subsets.
+        """
+        for dimension in dimensions:
+            # only look for a mask if there is a selected subset:
+            default_subset = getattr(self, dimension + '_subset').default_text
+            selected_subset = getattr(self, dimension + '_subset_selected')
+
+            if selected_subset != default_subset:
+                # if the selected subset is not the default, i.e.
+                # "Entire Spectrum" or "Entire Cube", retrieve masks:
+                if dimension == 'spectral':
+                    subset_mask = self.spectral_subset.selected_subset_mask
+                elif self.spatial_subset is not None:
+                    subset_mask = self.spatial_subset.selected_subset_mask
+                else:
+                    continue
```

Comment:
can this be simplified further (not sure I got all the indentation right, its hard to edit here)

```suggestion
        for subset_component in [self.spatial_subset, self.spectral_subset]:
            if subset_component is None:
                # spatial_subset only exists for cubeviz
                continue
           # only look for a mask if there is a selected subset:
           if subset_component.selected == subset_component.default_text:
               continue
            
           subset_mask = subset_component.selected_subset_mask
```

Reply:
I had to tweak the suggestion a bunch of times so I made a separate commit: 
255d5c5.